sdk/textanalytics/Azure.AI.TextAnalytics/src/Transforms.Legacy.cs

```diff
@@ -130,58 +134,51 @@ sentiment switch
                 _ => throw new NotSupportedException($"The document sentiment, { sentiment }, is not supported for conversion.")
             };
 
-        //TODO:  Check this conversion.  Seems to match: https://github.com/Azure/azure-sdk-for-net/blob/Azure.AI.TextAnalytics_5.2.0-beta.2/sdk/textanalytics/Azure.AI.TextAnalytics/src/DocumentSentiment.cs#L56
-        internal static List<SentenceOpinion> ConvertToSentenceOpinions(IReadOnlyList<Legacy.SentenceAssessment> assessments)
+        internal static List<SentenceOpinion> ConvertToSentenceOpinions(Legacy.SentenceSentiment currentSentence, IReadOnlyList<Legacy.SentenceSentiment> allSentences)
         {
-            var assessmentList = new List<AssessmentSentiment>(assessments.Count);
-            var targetList = new List<TargetSentiment>(assessments.Count);
+            var opinions = new List<SentenceOpinion>();
 
-            foreach (var assessment in assessments)
+            foreach (var target in currentSentence.Targets)
             {
-                targetList.Add(new TargetSentiment(
-                    ConvertToTextSentiment(assessment.Sentiment),
-                    assessment.Text,
-                    assessment.ConfidenceScores.Positive,
-                    assessment.ConfidenceScores.Negative,
-                    assessment.Offset,
-                    assessment.Length));
+                var assessments = new List<AssessmentSentiment>();
 
-                assessmentList.Add(new AssessmentSentiment(
-                    ConvertToTextSentiment(assessment.Sentiment),
-                    assessment.ConfidenceScores.Positive,
-                    assessment.ConfidenceScores.Negative,
-                    assessment.Text,
-                    assessment.IsNegated,
-                    assessment.Offset,
-                    assessment.Length));
-            }
+                foreach (var relation in target.Relations)
+                {
+                    if (relation.RelationType == Legacy.Models.TargetRelationType.Assessment)
+                    {
+                        assessments.Add(ResolveAssessmentReference(allSentences, relation.Ref));
+                    }
+                }
 
-            var opinions = new List<SentenceOpinion>(targetList.Count);
+                var sentiment = new TargetSentiment(
```

Comment:
nit: might be better to use `target`

Reply:
Target is used in the outer loop.   I changed to `targetSentiment`.